File: Livro-FundamentosDaProgramacaoDeComputadores/Capitulo3/Java/Exercicio24.java
/*24. Faça um programa que receba uma hora formada por hora e minutos (um número real), calcule e mostre a hora digitada apenas em minutos. Lembre-se de que:
 - para quatro e meia, deve-se digitar 4.30;
 - os minutos vão de 0 a 59.*/

import java.util.Scanner;

public class Exercicio24 {
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);

        System.out.print("Digite a hora: ");
        double hora = scan.nextDouble();

        double h = Math.floor(hora);
        double m = hora - h;
        double conversao = h * 60 + m * 100;

        System.out.printf("\nA hora convertida é igual a %.0f minutos.\n", conversao);

        scan.close();
    }
    
}

File: Livro-FundamentosDaProgramacaoDeComputadores/Capitulo3/C/Exercicio24.c
/*24. Faça um programa que receba uma hora formada por hora e minutos (um número real), calcule e mostre a hora digitada apenas em minutos. Lembre-se de que:
 - para quatro e meia, deve-se digitar 4.30;
 - os minutos vão de 0 a 59.*/

#include <stdio.h>
#include <math.h>

int main()
{
    float hora, h = 0, m = 0, conversao = 0;

    printf("Digite a hora: "); 
    scanf("%f", &hora);

    h = floor(hora);
    m = hora - h;
    conversao = (h * 60) + (m * 100);

    printf("A hora convertida é igual a %.0f minutos.", conversao);

    return 0;
}